docs/advanced/tls_hardware_acceleration.md

```diff
@@ -0,0 +1,131 @@
+# mbed TLS Hardware Acceleration
+
+This document explains how to use hardware acceleration for various cryptographic primitives with [mbed TLS](https://github.com/ARMmbed/mbedtls) running on an embedded device.
+
+
+## Introduction
+
+### Why should I add hardware acceleration?
+
+Whether the application developer uses mbed TLS as a cryptographic library or uses it as a TLS stack, they will end up with time-consuming cryptographic routines in their application that will heavily impact response time. The platform that reduces this adverse effect will have an edge over platforms that do not (or do so to a lesser extent).
+
+You may want to add hardware acceleration in the following cases:
+
+
+- You can accelerate parts significantly with optimised assembly code.
+
+
+- Your processor has special instructions capable of accelerating cryptographic operations.
+
+
+- Your processor has access to a co-processor with cryptographic acceleration capabilities.
+
+
+- Your platform has a dedicated crypto-module capable of executing cryptographic primitives, and possibly storing keys securely.
+
+The mbed TLS library was written in C and it doesn’t use any optimised assembly code. The only exception is the arbitrary precision multiplication on selected processors. You can find the list of supported platforms in the top comment in [bn_mul.h](https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/bn_mul.h).
```

Comment:
I think we can sound more positive here. Can I suggest:

'The mbed TLS library was written in C and it has a small amount of hand-optimised assembly code, limited to arbitrary precision multiplication on some processors.'